src/main/cpp/include/config/genomicsdb_config_base.h

```diff
@@ -164,6 +146,19 @@ class GenomicsDBConfigBase {
   }
   void scan_whole_array();
   const std::vector<std::string>& get_attributes() const { return m_attributes; }
+  //JSON parsing functions
```

Comment:
Would it be possible to push most of JSON parsing functions to json_config and have something like `void read_from_JSON_file(const std::string& filename, const int rank=0)`, basically the counterpart of `void read_from_PB(const genomicsdb_pb::ExportConfiguration* x, const int rank=0)`?

Reply:
Perhaps I'm misunderstanding this - do you need changes to the functions in lines 155-156 in this file?

Reply:
I re-read this - are you suggesting moving the static parsing functions to json_config away from the GenomicsDBConfigBase class? That's a good idea

Reply:
Done

Reply:
Thanks Karthik. Moving the static parsing functions to json_config is great.  Do we need both `read_from_file() - line 150` and `read_from_JSON_string() - line 151`? Also, what I meant was - do we need any reference to rapidjson in GenomicsDBConfigBase at all? Can the reference be relegated to json_config.h/json_config.cc?

Reply:
The only reason that those functions return a rapidjson::Document object is that GenomicsDBImportConfig also calls GenomicsDBConfigBase::read_from_filename to parse the common fields and then uses the rapidjson::Document object returned to read in the other import specific fields.